third_party/blink/renderer/core/dom/events/event_target.cc

```diff
@@ -723,6 +724,30 @@ bool EventTarget::dispatchEventForBindings(Event* event,
     return false;
 
   event->SetTrusted(false);
+  
+  if (recordreplay::IsRecordingOrReplaying()) {
+    if (event->IsMouseEvent() &&
+      (event->type() == event_type_names::kMouseup ||
+      event->type() == event_type_names::kMousedown ||
+      event->type() == event_type_names::kMousemove)
+    ) {
+      // Node* node = ToNode();
+      // gfx::Rect bbox = node ? node->PixelSnappedBoundingBox() : gfx::Rect();
+      // gfx::Point center = bbox.CenterPoint();
+      auto& mouseEvent = To<MouseEvent>(*event);
+      size_t x = (size_t)std::round(mouseEvent.clientX());
+      size_t y = (size_t)std::round(mouseEvent.clientY());
+      recordreplay::OnMouseEvent(event->type().Utf8().c_str(), x, y, true);
+    } else if (event->IsKeyboardEvent() &&
+      (event->type() == event_type_names::kKeydown ||
+      event->type() == event_type_names::kKeyup ||
+      event->type() == event_type_names::kKeypress)
+    ) {
+      recordreplay::OnKeyEvent(event->type().Utf8().c_str(),
+                               DynamicTo<KeyboardEvent>(event)->key().Utf8().c_str(),
```

Comment:
why `DynamicTo<>` here but `To<>` for mouse events?  if it's necessary might be worth commenting as why.  Otherwise switch to `To<>`?

Reply:
Whoops.